src/pages/2023-12/next-js-app-router-cache/index.mdx

```diff
@@ -0,0 +1,388 @@
+
+Whether we choose to opt-out of data cache or not, the requests will still be memoized to avoid **duplicate requests**.
+
+##### Revalidation
+
+While data cache is great, there are times where we would want to revalidate it, or disable it altogether.
+
+If we look at the code above, what happens when a new guide is published on the topic of "best cafes in Paris"? Since there is no revalidation at the moment, the subsequent users who visit the site will see stale data.
+
+Next.js provides us two ways to revalidate the data cache: time-based revalidation and on-demand revalidation.
+
+**Time-based revalidation** is useful when we want to make a new request after a certain amount of time has passed. This is useful if the data in the data source changes frequently.
+
+This would change our above `fetch` request to the following:
+
+```jsx
+// Revalidate at most every hour
+const res = fetch(`https://api.globetrotter.com/guides/${city}`, {
+  next: { revalidate: 3600 },
+});
+```
+
+The code above will revalidate the request every hour. Meaning, if the time difference between two requests is more than an hour, the cache becomes stale.
+
+**On-demand revalidation** is useful when data is required to be refreshed based on a specific event. In our case, we want to revalidate the cache when there is a new post in the blog.
+
+The code below demonstrates on-demand revalidation. Every time `publishArticle` is called, Next.js will attempt to regenerate `/guides/paris`.
+
+```jsx {10}
+import { revalidatePath } from "next/cache";
+
+export async function publishArticle() {
+  try {
+    // ...
+  } catch (error) {
+    // ...
+  }
+
+  revalidatePath("/guides/paris");
+}
+```
+
+`revalidatePath()` will purge all the server side cache for the `/guides/paris` route. This means the next user to make a request will recieve the most up to date data, fetched directly from the source.
+
+##### Opting out
+
+To opt-out, we simply use the `{cache:'no-cache'}` option. This means all the `fetch` requests will go directly to the data source, effectively disabling the data cache.
+
+```jsx
+// Revalidate at most every hour
+const res = fetch(`https://api.globetrotter.com/guides/${city}`, {
+  cache: "no-cache",
+});
+```
+
+When might we want to disable the data cache?
+
+1. **Real-time Data:** The information changes frequently and must always be up-to-date, e.g. stock trading platform where users expect to see real-time price updates.
+2. **User-specific Data:** The fetched data is unique to each user session and should not be shared, e.g. personalized dashboard showing private messages.
+3. **Security:** Caching may pose a security risk, especially if sensitive data is involved and improperly isolated, e.g. ana pplication that handles sensitive medical records.
+
+The diagram below demonstrates a step-by-step process of how data cache works.
+
+![data-cache](/articleAssets/2023-12/next-js-app-router-cache/data-cache.png)
+
```

Comment:
There is nothing in this section about multiple topics:
1. `unstable_cache` from Next.js
2. `revalidateTag`
3. `export const dynamic`
4. `export const revalidate`